BaragonService/src/main/java/com/hubspot/baragon/service/elb/ApplicationLoadBalancer.java

```diff
@@ -29,23 +62,413 @@ public ApplicationLoadBalancer(Optional<ElbConfiguration> configuration,
     this.elbClient = elbClient;
   }
 
-  public boolean isInstanceHealthy(String instanceId, String name) {
-    return true;
-  }
+  @Override
+  public boolean isInstanceHealthy(String instanceId, String elbName) {
+    DescribeTargetHealthRequest healthRequest = new DescribeTargetHealthRequest()
+        .withTargets(new TargetDescription().withId(instanceId));
+    DescribeTargetHealthResult result = elbClient.describeTargetHealth(healthRequest);
 
-  public Optional<LoadBalancer> getElb(String elbName) {
-    return Optional.absent();
+    for (TargetHealthDescription health: result.getTargetHealthDescriptions()) {
+      if (health.getTargetHealth().getState().equals(TargetHealthStateEnum.Healthy.toString())
+          && health.getTarget().getId().equals(instanceId)) {
+        return true;
+      }
+    }
+
+    return false;
   }
 
+  @Override
   public void removeInstance(Instance instance, String elbName, String agentId) {
-
+    /* TODO: Need to map v1 Instance -> v2 Target */
   }
 
+  @Override
   public RegisterInstanceResult registerInstance(Instance instance, String elbName, BaragonAgentMetadata agent) {
+    /* TODO: Need to map v1 Instance -> v2 Target */
     return RegisterInstanceResult.ELB_AND_VPC_FOUND;
   }
 
-  public void syncAll(Collection<BaragonGroup> groups) {
+  @Override
+  public void syncAll(Collection<BaragonGroup> baragonGroups) {
+    Collection<LoadBalancer> allLoadBalancers = getAllLoadBalancers(baragonGroups);
+    try {
+      for (BaragonGroup baragonGroup : baragonGroups) {
```

Comment:
Since a few of the methods in this for loop can possibly throw exceptions, we should probably move the try/catch _inside_ the loop. This way we are still syncing the remaining groups, even if one should fail